invenio_github/api.py

```diff
@@ -301,165 +394,147 @@ def check_token(cls, token):
-
-    @cached_property
-    def description(self):
-        """Extract description from a release."""
-        if self.release.get('body'):
-            return markdown(self.release['body'])
-        elif self.repository.get('description'):
-            return self.repository['description']
-        return 'No description provided.'
-
-    @cached_property
-    def author(self):
-        """Extract the author's GitHub username from a release."""
-        return self.release.get('author', {}).get('login')
-
-    @cached_property
-    def related_identifiers(self):
-        """Yield related identifiers."""
-        yield dict(
-            identifier=u'https://github.com/{0}/tree/{1}'.format(
-                self.repository['full_name'], self.release['tag_name']
-            ),
-            relation='isSupplementTo',
-        )
+        return self.payload["repository"]
 
     @cached_property
-    def version(self):
-        """Extract the vesion from the release tag."""
-        return self.release.get('tag_name', '')
+    def repository_object(self):
+        """Return repository model from database."""
+        if self.release_object.repository_id:
+            repository = self.release_object.repository
+        else:
+            repository = Repository.query.filter_by(
+                user_id=self.event.user_id,
+            ).one()
+        return repository
 
     @cached_property
-    def defaults(self):
-        """Return default metadata."""
-        return dict(
-            access_right='open',
-            description=self.description,
-            license='other-open',
-            publication_date=self.release['published_at'][:10],
-            related_identifiers=list(self.related_identifiers),
-            version=self.version,
-            title=self.title,
-            upload_type='software',
-        )
+    def release_file_name(self):
+        """Returns release zipball file name."""
+        tag_name = self.release_payload["tag_name"]
+        repo_name = self.repository_payload["full_name"]
+        filename = f"{repo_name}-{tag_name}.zip"
+        return filename
 
     @cached_property
-    def extra_metadata(self):
-        """Get extra metadata for file in repository."""
-        return get_extra_metadata(
-            self.gh.api,
-            self.repository['owner']['login'],
-            self.repository['name'],
-            self.release['tag_name'],
-        )
+    def release_zipball_url(self):
+        """Returns the release zipball url."""
+        return self.release_payload["zipball_url"]
 
     @cached_property
-    def citation_metadata(self):
-        """Get citation metadata for file in repository."""
-        return get_citation_metadata(
-            self.gh.api,
-            self.repository['owner']['login'],
-            self.repository['name'],
-            self.release['tag_name'],
-            self.model
-        )
-
+    def user_identity(self):
```

Comment:
This feels a bit out of place here. 

The problem was that to publish a record, we needed the user's identity. Therefore, we used a utility function from invenio-access to generate this identity given the user id. 

But, there is permission needed to publish a draft (`authenticated_user`) and it was added here.

Possible alternatives:

1 - move this to `invenio-rdm-records` as this is a specific RDM action (publish)
2 - move this to `invenio-access` as a new utility function (e.g. `get_authenticated_identity`)

Reply:
Created an [issue](https://github.com/inveniosoftware/invenio-github/issues/105)